### PR TITLE
[improve][client] Add backoff for `seek`

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation
Currently, only `seek` doesn't support backoff in consumer side, see :
https://github.com/apache/pulsar/blob/2ab184e49a036a1dd10dc537bef4ab034a5ad5e0/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java#L2151-L2166

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java [hunk 1]
```diff
@@ -2148,100 +2148,108 @@ public CompletableFuture<Void> seekAsync(Function<String, Object> function) {
...
-            clearIncomingMessages();
-            seekFuture.complete(null);
-        }).exceptionally(e -> {
-            // re-set duringSeek and seekMessageId if seek failed
-            seekMessageId = originSeekMessageId;
-            duringSeek.set(false);
-            log.error("[{}][{}] Failed to reset subscription: {}", topic, subscription, e.getCause().getMessage());
+                clearIncomingMessages();
+                seekFuture.complete(null);
+            }).exceptionally(e -> {
```
Comment (codelipenghui) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java:2190]: If the exception is retriable, I think we should also perform a backoff.

**Review by codelipenghui — COMMENTED** [review]
Is it possible to add a test to cover the new changes?

**Comment by Technoboy-** [comment]
> Is it possible to add a test to cover the new changes?

Added test.

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java (hunk 1 above)
Comment (Technoboy-) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java:2190]: Ah, the server side will response with `MetadataError/UnknownError`, which we can't retry. I see other methods here, also not handle exception to take a retry.

**Review by codelipenghui — APPROVED** [review]

**Review by coderzc — APPROVED** [review]

**Review by Demogorgon314 — APPROVED** [review]

**Pull request merged**